python/ray/tests/runtime_env_container/test_ray_env_vars.py

```diff
@@ -6,17 +6,28 @@
 
 parser = argparse.ArgumentParser()
 parser.add_argument("--image", type=str, help="The docker image to use for Ray worker")
+parser.add_argument(
+    "--use-image-uri-api",
+    action="store_true",
+    help="Whether to use the new `image_uri` API instead of the old `container` API.",
+)
 args = parser.parse_args()
 
 worker_pth = get_ray_default_worker_file_path()
 
 
-@ray.remote(runtime_env={"container": {"image": args.image, "worker_path": worker_pth}})
+if args.use_image_uri_api:
+    runtime_env = {"image_uri": args.image}
+else:
+    runtime_env = {"container": {"image": args.image, "worker_path": worker_pth}}
+
+
+@ray.remote(runtime_env=runtime_env)
 def f():
     return os.environ.get("RAY_TEST_ABC")
 
 
-@ray.remote(runtime_env={"container": {"image": args.image, "worker_path": worker_pth}})
+@ray.remote(runtime_env=runtime_env)
 def g():
     return os.environ.get("TEST_ABC")
```

Comment:
Note: on `container` this is expected to NOT be inherited. However on `image_uri` maybe we should inherit it? since we support `env_vars`.